roles/swift-ring/tasks/main.yml

```diff
@@ -13,3 +13,4 @@
   command: sudo -u swiftops /usr/local/bin/swifttool -i /home/swiftops/.ssh/id_rsa
            bootstrap --config /etc/swift/ring_definition.yml --outdir /etc/swift
   when: not should_boostrap|success
+  notify: restart swift services
```

Comment:
https://github.com/blueboxgroup/ursula/blob/master/roles/openstack-source/tasks/main.yml#L22  is the bit of code that takes the supplied python requirements when calling the role and making them show up inside the virtualenv.